### PR TITLE
#1180 : rename fmi3NewDiscreteStates to fmi3UpdateDiscreteStates

### PR DESCRIPTION
**Pull request by masoud-najafi**
Following the decision on renaming fmi3NewDiscreteStates  to fmi3UpdateDiscreteStates (https://github.com/modelica/fmi-standard/issues/1180) , I renamed all occurrences of fmi3NewDiscreteStates.
We need also update the APIs in reference FMUs.
CC: @t-sommer

### CODE REVIEW

**Review by chrbertsch — APPROVED** [review]

**Review by andreas-junghanns — CHANGES_REQUESTED** [review]
`newDiscreteStatesNeeded` is still around - should we not also change that argument accordingly?

**Comment by masoud-najafi** [comment]
> 
> 
> `newDiscreteStatesNeeded` is still around - should we not also change that argument accordingly?

If "updateDiscreteStatesNeeded" is fine, I will replace them all.

**Comment by chrbertsch** [comment]
Perhaps  add "of": updateOfDiscreteStatesNeeded"?

**Comment by andreas-junghanns** [comment]
> 
> 
> Perhaps add "of": updateOfDiscreteStatesNeeded"?

I think without "of" is better, because updateDiscreteStates is the function call, updateDiscreteStatesNeeded requires another call to that function. And it would be shorter too.

File: headers/fmi3FunctionTypes.h [hunk 1]
```diff
@@ -488,15 +488,15 @@ typedef fmi3Status fmi3SetIntervalFractionTYPE(fmi3Instance instance,
                                                size_t nValues);
 /* end::SetIntervalFraction[] */
 
-/* tag::NewDiscreteStates[] */
-typedef fmi3Status fmi3NewDiscreteStatesTYPE(fmi3Instance instance,
-                                             fmi3Boolean *newDiscreteStatesNeeded,
+/* tag::UpdateDiscreteStates[] */
+typedef fmi3Status fmi3UpdateDiscreteStatesTYPE(fmi3Instance instance,
```
Comment (t-sommer) [inline headers/fmi3FunctionTypes.h:492]: Shouldn't this be `discreteStatesNeedUpdate` instead of `updateDiscreteStatesNeeded`?

**Review by t-sommer — CHANGES_REQUESTED** [review]

**Review by t-sommer — APPROVED** [review]

**Review by chrbertsch — APPROVED** [review]

**Review by andreas-junghanns — APPROVED** [review]

**Pull request merged**